### PR TITLE
RR-227 - Introduced client and service methods to be able to call the createActionPlan API

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**
This PR introduces the client and service methods to be able to call the `createActionPlan` API (rather than the UI calling the `createGoals` method when it wants to create the action plan)

This PR only introduces the methods, it does not wire them in or use them.

My next PR will wire the call to the service method into the relevant points in the user journey 👍

### CODE REVIEW

File: server/@types/educationAndWorkPlanApi/index.d.ts [hunk 1]
```diff
@@ -4,6 +4,54 @@
  */
 
 export interface paths {
+  '/queue-admin/retry-dlq/{dlqName}': {
```
Comment (nathanrussell-moj-digital) [inline server/@types/educationAndWorkPlanApi/index.d.ts:7]: Re-imported the swagger spec, which has brought in with it lots of unrelated change such as the DLQ endpoints etc.

The change we really wanted to bring in by importing the swagger spec was the removal of the `reviewDate` field, which is below 👇

File: server/data/educationAndWorkPlanClient.ts [hunk 2]
```diff
@@ -30,16 +31,27 @@ export default class EducationAndWorkPlanClient {
...
-      data: createGoalsRequest,
+  async getActionPlan(prisonNumber: string, token: string): Promise<ActionPlanResponse> {
+    return EducationAndWorkPlanClient.restClient(token).get<ActionPlanResponse>({
+      path: `/action-plans/${prisonNumber}`,
     })
   }
 
-  async getActionPlan(prisonNumber: string, token: string): Promise<ActionPlanResponse> {
-    return EducationAndWorkPlanClient.restClient(token).get<ActionPlanResponse>({
+  async createActionPlan(
```
Comment (nathanrussell-moj-digital) [inline server/data/educationAndWorkPlanClient.ts:40]: The diff looks bigger than it is because I moved/re-ordered the methods `getActionPlan` and createGoals`; but really what we have here is the introduction of the new `createActionPlan` method

**Review by stevomcallister — APPROVED** [review]
